pennylane/operation.py

```diff
@@ -1291,23 +1291,15 @@ def parameter_frequencies(self):
         if self.num_params == 1:
             # if the operator has a single parameter, we can query the
             # generator, and if defined, use its eigenvalues.
-            gen = self.generator()
 
-            try:
-                gen_eigvals = tuple(self.generator().eigvals())
-                return qml.gradients.eigvals_to_frequencies(gen_eigvals)
-
-            except (MatrixUndefinedError, EigvalsUndefinedError):
-
-                if isinstance(gen, qml.Hamiltonian):
-                    mat = qml.utils.sparse_hamiltonian(gen).toarray()
-                    eigvals = tuple(np.round(np.linalg.eigvalsh(mat), 8))
-                    return qml.gradients.eigvals_to_frequencies(eigvals)
+            with warnings.catch_warnings():
+                warnings.filterwarnings(
+                    action="ignore", message=r".+ eigenvalues will be computed numerically\."
+                )
+                eigvals = qml.eigvals(qml.generator(self, format="observable"))
 
-                if isinstance(gen, qml.SparseHamiltonian):
-                    mat = gen.sparse_matrix().toarray()
-                    eigvals = tuple(np.round(np.linalg.eigvalsh(mat), 8))
-                    return qml.gradients.eigvals_to_frequencies(eigvals)
+            eigvals = tuple(np.round(eigvals, 8))
+            return qml.gradients.eigvals_to_frequencies(eigvals)
```

Comment:
@dwierichs I made the change here because I realized the old version was simply duplicating `qml.eigvals()`; this makes the method cleaner, and less likely to break in the future.

Reply:
Oh nice, this _does_ look very nice :)